server/src/main/java/org/elasticsearch/search/aggregations/metrics/AbstractInternalTDigestPercentiles.java

```diff
@@ -24,6 +25,10 @@
 
 abstract class AbstractInternalTDigestPercentiles extends InternalNumericMetricsAggregation.MultiValue {
 
+    // NOTE: using compression = 1.0 empty histograms will track just about 5 centroids.
+    // This reduces the amount of data to serialize and deserialize.
+    private static final TDigestState EMPTY_HISTOGRAM = new TDigestState(1.0D);
```

Comment:
How much memory does `EMPTY_HISTOGRAM` take? Or what is the size of the underlying array(s)?

Reply:
`AvlTreeDigest` JavaDoc says that, if comrpession = N then `you should expect to track about 5 N centroids for this accuracy`. In AvlTree each node is a centroid and the AvlTree uses an array as the udnerlying data structure with an array that has always M elements where M > number of tree nodes.

Reply:
Look like the empty instance has one instance of `IntAVLTree` with 3 int arrays and 1 byte arrays? All arrays are initialised with length 16. The single instance is only used for xcontent serialization. So I think it never grows beyond that.

Reply:
There are also two int arrays and double array in `AVLGroupTree`. Both are initialised based on size of `IntAVLTree` which is 16.